master/docs/manual/customization.rst

```diff
@@ -539,26 +539,34 @@ Aside from the service methods, the other concerns in the previous section apply
 Writing a New Latent Worker Implementation
 ------------------------------------------
 
-Writing a new latent worker should only require subclassing :class:`buildbot.worker.AbstractLatentWorker` and implementing :meth:`start_instance` and :meth:`stop_instance`.
+Writing a new latent worker should only require subclassing :class:`buildbot.worker.AbstractLatentWorker` and implementing :meth:`start_instance` and :meth:`stop_instance` at a minimum.
 
-::
+.. bb:worker:: AbstractWorkerController
+
+AbstractLatentController
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-    def start_instance(self):
-        # responsible for starting instance that will try to connect with this
-        # master. Should return deferred. Problems should use an errback. The
-        # callback value can be None, or can be an iterable of short strings to
-        # include in the "substantiate success" status message, such as
-        # identifying the instance that started.
-        raise NotImplementedError
+.. py:class:: buildbot.worker.AbstractLatentWorker
 
-    def stop_instance(self, fast=False):
-        # responsible for shutting down instance. Return a deferred. If `fast`,
-        # we're trying to shut the master down, so callback as soon as is safe.
-        # Callback value is ignored.
-        raise NotImplementedError
+This class is the base class of all latent workers and implements some common functionality.
+A custom worker should only need to override :meth:`start_instance` and :meth:`stop_instance` methods.
 
 See :class:`buildbot.worker.ec2.EC2LatentWorker` for an example.
```

Comment:
The EC2 worker is one of the first latent workers and has grown in complexity to be the largest latent worker. I think we should find a better example.

The Hyper worker might be better since @tardyp wrote it recently. It's the second shortest after Marathon and includes putting any `hyper_sh` method calls in a thread pool to handle them being blocking calls.

Reply:
Agreed, and also https://github.com/buildbot/buildbot/pull/4419 implements the new API only on hyper.